src/pages/home/library/EditSong.tsx

```diff
@@ -201,43 +203,33 @@ const EditSong: FunctionComponent = () => {
   }
 
   const handleSubmit = async (
+    step: "basic-details" | "advanced-details" | "confirm",
     values: PatchSongRequest,
     helpers: FormikHelpers<FormikValues>
   ) => {
-    await patchSong({ ...values, shouldRedirect: true });
-    helpers.setSubmitting(false);
-  };
-
-  // Navigate to advanced details if minting, otherwise upload song
-  const handleSongInfo = async (
-    values: PatchSongRequest,
-    helpers: FormikHelpers<FormikValues>
-  ) => {
-    if (values.isMinting) {
-      await patchSong({
-        ...values,
-        isMinting: false,
-      });
-
-      helpers.setSubmitting(false);
+    const patchValues = {
+      ...values,
+      isMinting: false,
+      shouldRedirect: false,
+    };
 
-      navigate("advanced-details");
-    } else {
-      await handleSubmit(values, helpers);
+    if (step === "basic-details") {
+      patchValues.isMinting = !values.isMinting;
```

Comment:
I believe this is causing the patchSong thunk to execute the body.isMinting conditional 
`      if (body.isMinting) {` within `src/modules/song/thunks.ts` on line 418, which dispatches unnecessarily for edit song "basic details" updates, e.g. `generateArtistAgreement` is triggered for a basic save.

I think it would be fixed with just removing the line since it is false by default and the flow will still work, but if you see that it is needed within the logic just adding an additional statement or add an `else` for `body.shouldRedirect` would work: 
```     if (body.isMinting && !body.shouldRedirect) {```